### PR TITLE
[Fix](multi-catalog) Fix query hms tbl with compressed data files.

### PR DESCRIPTION
**Pull request by dutyu**
# Proposed changes

Issue Number: close #xxx

## Problem summary

If submit a query contains hms tbls which data files are compressed (bz2,lzo,lz4 ...), a error will occurs like this: 

```[INTERNAL_ERROR]Only support csv data in utf8 codec``` . 

This is because `org.apache.doris.planner.external.HiveScanNode`  set `fileFormatType` as `TFileFormatType.FORMAT_CSV_PLAIN` whether the real compress algo of data files are.  This pr try to fix this problem.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by dutyu** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.59 seconds
 stream load tsv:          417 seconds loaded 74807831229 Bytes, about 171 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          60 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230509033835_clickbench_pr_140614.html

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/FileQueryScanNode.java [hunk 1]
```diff
@@ -211,9 +212,9 @@ public void createScanRangeLocations() throws UserException {
         FileSplit inputSplit = (FileSplit) inputSplits.get(0);
         TFileType locationType = getLocationType();
         params.setFileType(locationType);
-        TFileFormatType fileFormatType = getFileFormatType();
-        params.setFormatType(getFileFormatType());
-        if (fileFormatType == TFileFormatType.FORMAT_CSV_PLAIN || fileFormatType == TFileFormatType.FORMAT_JSON) {
+        TFileFormatType fileFormatType = getFileFormatType(inputSplit);
+        params.setFormatType(fileFormatType);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/FileQueryScanNode.java:216]: Better use `params.setCompressType()` to set compress type separately

**Comment by dutyu** [comment]
run buildall

**Comment by dutyu** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**